pkg/datapath/loader/base.go

```diff
@@ -158,6 +160,24 @@ func addENIRules(sysSettings []sysctl.Setting) ([]sysctl.Setting, error) {
 		return nil, fmt.Errorf("unable to install ip rule for ENI multi-node NodePort: %w", err)
 	}
 
+	// Add rules for router (cilium_host).
+	info := node.GetRouterEniInfo()
+	cidrs := info.IPv4CIDRs
+	routerIP := net.IPNet{
+		IP:   nodeAddressing.IPv4().Router(),
+		Mask: net.CIDRMask(32, 32),
+	}
+	for _, cidr := range cidrs {
+		if err := route.ReplaceRule(route.Rule{
+			Priority: linux_defaults.RulePriorityEgressv2,
```

Comment:
This will actually depend on whether `egress-multi-home-ip-rule-compat` is set or not. It defaults to false. True would mean that we want to maintain compatibility with the older ENI datapath (pre https://github.com/cilium/cilium/pull/14269)

Reply:
Right, ideally these particular details (table /priority) should be handled in the linuxrouting package and this package should just ask for the table number.

Maybe this entire logic should just be pushed to a new function like `SetupRoutes()` exposed from `linuxrouting`? That way all of the logic around priority, table, RouterInfo introspection can be handled in just that one package rather than spreading it also to this package which must now take care of nuanced details of the routes.

Reply:
That sounds like a reasonable idea to me. Much better than having seemingly random logic scattered around the codebase regarding rules and such.

Reply:
Agreed. 
>This will actually depend on whether egress-multi-home-ip-rule-compat is set or not.

I don't think I follow this. AFAIU, this flag comes into play only during upgrade? But in this case, we were never installing the `cilium_host` rules before?

Reply:
The flag determines priority and table number for the rules, so this code should be consistent with which approach is used to determining these two values. The concern is if this isn't consistent, then the new rule which we install here may point to the wrong route table.

Reply:
>The flag determines priority and table number for the rules, so this code should be consistent with which approach is used to determining these two values.

 Ah ok. Prior to v1.9 (or whichever release this flag was introduced in), the ip rules were installed in table Idx so we want to install the new rules to the same table Id.

Reply:
FWIW the related PR is https://github.com/cilium/cilium/pull/14269 and it is still being backported. Both modes should be supported for now (until we deprecate & remove the flag).